src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -255,6 +268,76 @@ public void Initialize(IEventSource eventSource)
         }
     }
 
+    /// <summary>
+    /// Parses out the logger parameters from the Parameters string.
+    /// </summary>
+    public void ParseParameters()
+    {
+        var parameters = LoggerParametersHelper.ParseParameters(Parameters);
+
+        foreach (var parameter in parameters)
+        {
+            ApplyParameter(parameter.Key, parameter.Value);
+        }
+    }
+
+    /// <summary>
+    /// Apply a terminal logger parameter.
+    /// parameterValue may be null, if there is no parameter value.
+    /// </summary>
+    private bool ApplyParameter(string parameterName, string? parameterValue)
```

Comment:
The value meant whether the function was successful or not. Should have called it `TryApplyParameter`.
The idea was not to throw on parse error and communicate the success this way. But since we anyway ignore this value, the class is sealed and the function is not public, we can change it anytime if we need such info for our logic, so I will just remove this bool.